docs/src/postmortem/2021-07-17.md

```diff
@@ -0,0 +1,114 @@
+relies on TLS certificates.  This certificate expired without being
+noticed which prevented what would have otherwise been a quick
+recovery of logging in and bouncing some services.
+
+Additionally, when the CAS is unavailable, we maintain a break-glass
+login capability for a handful of extremely trusted users (1:1 with
+the people that have access to the package signing key).  This access
+was discovered to be impaired by 1 developer's key being missing
+entirely, and one developer having failed to rotate their key.  The
+third developer was on vacation but was able to log in and rectify the
+keying problem.
+
+## Fault
+
+The Nomad outage was caused by an unexpected restart of a-fsn-de.
+When Nomad hosts reboot there is a known defect in that runit may
+bring up the services in a race that results in Nomad not being usable
+until services are restarted in a specific order.
+
+The unavailability of the CAS system means that we still cannot log in
+to all hosts normally.
+
+The issues with the break-glass keys caused the recovery of both the
+specific Nomad host and the CAS server to be slow.
+
+## Impact
+
+Publicly Visible:
+
+  * Repository signing unavailable
+  * Builds not completing normally
+  * musl and aarch64 appeared to lag behind glibc
+
+Internally Visible:
+
+  * CAS logins not available
+  * Detailed failure logs not available in Grafana (requires CAS login)
+  * Couldn't make control requests to Nomad (requires Vault CAS integration)
+
+## Detection
+
+The Nomad failure was detected by external observation that an update
+the the `less` package was not signed, and was preventing installs
+from proceeding.
+
+The CAS failure was discovered a few days before the signing issue but
+was deemed non-critical as it was an invonvenience that could be fixed
+within a week.
+
+## Response
+
+The ability to merge changes in GitHub was restricted to prevent new
+builds from running that might further complicate recovery efforts.
+
+@the-maldridge and @Gottox were recalled from vacation to recover
+access to the system.
+
+## Recovery
+
+@Gottox used break-glass access to both restore break-glass access for
+@the-maldridge and @duncaen, and to restart the stuck signing process.
+
+The CAS failure has not been resolved at the time of publication.
+
+
+## What went well
+
+  * Excellent internal communication kept everyone in the loop as to
+    what was broken, what was being done to fix it, and who was
+    responsible for taking action.
+
+  * Break-glass access, once used, did work effectively.
+
+
+## What could be done better
+
+  * External communication was not great.  A post went up on
+    voidlinux.org, but no twitter notification was made, and the post
+    was not widely shared on our other channels.
+
+  * Break glass connectivity existed, but did not work.
+
+  * Initially recalling critical team members from vacation was an
+    ad-hoc process.
+
+
+## Lessons learned
```

Comment:
I would add that 1) a better pipeline to only publish packages once signed and/or also sign repodata would have prevented the users from realizing the issue, but might have delayed detection (we took a while to realize the builder issue was in signing instead of syncing, iirc); so we should also look into improving monitoring and warnings.

Not sure if fits under "done better" or "lessons learned".